### PR TITLE
Add kms hsm support for signing

### PR DESCRIPTION
**Pull request by vcastellm**
This PR adds support for signing L1 transactions using GCP KMS.

In case no GCP Key is indicated it fallbacks to using locally stored private keys.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -90,9 +87,39 @@ func start(cliCtx *cli.Context) error {
...
 
 	// Prepare Etherman
-	auth, err := bind.NewKeyedTransactorWithChainID(pk, big.NewInt(c.L1.ChainID))
-	if err != nil {
-		log.Fatal(err)
+
+	// Load private key
+	var auth *bind.TransactOpts
+	var addr common.Address
+	if c.KMSKeyName != "" {
```
[the remark this answers is not in the document]
Comment (vcastellm) [reply cmd/main.go:94]: Done in https://github.com/0xPolygon/agglayer/pull/67/commits/8a88768f9ba6c476f92dbb8eb559017d661b2d21

File: cmd/main.go [hunk 2]
```diff
@@ -90,9 +87,39 @@ func start(cliCtx *cli.Context) error {
...
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
[the remark this answers is not in the document]
Comment (vcastellm) [reply cmd/main.go]: yes but, what's the timeout configured in the cli context?

File: cmd/main.go [hunk 3]
```diff
@@ -72,56 +76,62 @@ func start(cliCtx *cli.Context) error {
...
 
 	// Prepare Etherman
-	auth, err := bind.NewKeyedTransactorWithChainID(pk, big.NewInt(c.L1.ChainID))
-	if err != nil {
-		log.Fatal(err)
+
+	// Load private key
+	var auth *bind.TransactOpts
+	var addr common.Address
+	if c.KMSKeyName != "" {
```
Comment (Stefan-Ethernal) [inline cmd/main.go:94]: Nitpick:
```suggestion
	var (
		auth *bind.TransactOpts
		addr common.Address
	)
```

File: cmd/main.go [hunk 4]
```diff
@@ -72,56 +76,62 @@ func start(cliCtx *cli.Context) error {
...
+		}
+	} else {
+		return errors.New("no private key found")
 	}
 
 	// Connect to ethereum node
 	ethClient, err := ethclient.DialContext(cliCtx.Context, c.L1.NodeURL)
 	if err != nil {
-		log.Fatal("error connecting to %s: %+v", c.L1.NodeURL, err)
+		return fmt.Errorf("error connecting to %s: %+v", c.L1.NodeURL, err)
```
Comment (Stefan-Ethernal) [inline cmd/main.go:111]: Shall we use %w?
```suggestion
		return fmt.Errorf("error connecting to %s: %w", c.L1.NodeURL, err)
```

File: cmd/main.go [hunk 5]
```diff
@@ -72,56 +76,62 @@ func start(cliCtx *cli.Context) error {
...
 	}
 
 	// Make sure the connection is okay
 	if _, err = ethClient.ChainID(cliCtx.Context); err != nil {
-		log.Fatal("error getting chain ID from l1 with address: %+v", err)
+		return fmt.Errorf("error getting chain ID from l1 with address: %+v", err)
```
Comment (Stefan-Ethernal) [inline cmd/main.go:116]: Shall we use `%w`?

```suggestion
		return fmt.Errorf("error getting chain ID from l1 with address: %w", err)
```

File: config/config.go [hunk 6]
```diff
@@ -34,6 +34,7 @@ type Config struct {
 	EthTxManager ethtxmanager.Config `mapstructure:"EthTxManager"`
 	L1           L1Config            `mapstructure:"L1"`
 	Telemetry    Telemetry           `mapstructure:"Telemetry"`
+	KMSKeyName   string
```
Comment (Stefan-Ethernal) [inline config/config.go:37]: How this config value gets populated? Do we need `mapstructure` for it as well if it is populated from the `aggalayer.toml`?

Comment (vcastellm) [reply config/config.go]: Moved to ethtxmanager config

**Comment by vcastellm** [comment]
Applied feedback

**Review by Stefan-Ethernal — APPROVED** [review]

**Pull request merged**